c/tskit/trees.c

```diff
@@ -1836,6 +1904,249 @@ tsk_treeseq_sample_count_stat(tsk_treeseq_t *self,
     return ret;
 }
 
+/***********************************
+ * Allele frequency spectrum
+ ***********************************/
+
+static int
+update_site_jafs(tsk_site_t *site,
+        double *total_counts, double *counts, tsk_size_t num_sample_sets,
+        tsk_size_t window_index, tsk_size_t *result_dims, double *result,
+        tsk_flags_t options)
+{
+    int ret = 0;
+    size_t j, afs_size;
+    tsk_size_t k, c, allele, num_alleles;
+    double increment, sum, *afs, *allele_counts, *allele_count;
+    tsk_size_t *coordinate = malloc(num_sample_sets * sizeof(*coordinate));
+    bool polarised = !!(options & TSK_STAT_POLARISED);
+
+    if (coordinate == NULL) {
+        ret = TSK_ERR_NO_MEMORY;
+        goto out;
+    }
+    ret = get_allele_weights(site, counts, num_sample_sets, total_counts, &num_alleles,
+            &allele_counts);
+    if (ret != 0) {
+        goto out;
+    }
+
+    afs_size = 1;
+    for (j = 0; j < num_sample_sets; j++) {
+        afs_size *= result_dims[j];
```

Comment:
shouldn't `afs_size` be passed in, at least so it isn't re-computed for every site?

Reply:
Yes, that would probably help.